src/tools/x/src/main.rs

```diff
@@ -132,6 +128,35 @@ fn main() {
             handle_result(result, cmd);
         }
     }
+}
+
+pub fn find_and_run_py_bootstrap_script() {
+    let current_path = get_current_dir();
+
+    for dir in current_path.ancestors() {
+        let candidate = dir.join("x.py");
+        if candidate.exists() {
+            let mut cmd: Command;
+            cmd = Command::new(python());
+            cmd.arg(&candidate).args(env::args().skip(1)).current_dir(dir);
+            let result = exec_or_status(&mut cmd);
+            handle_result(result, cmd);
+        }
+    }
+}
+
+#[allow(dead_code)]
+fn main() {
+    match env::args().skip(1).next().as_deref() {
+        Some("--wrapper-version") => {
+            let version = env!("CARGO_PKG_VERSION");
+            println!("{}", version);
+            return;
+        }
+        _ => {}
+    }
```

Comment:
Do we want this for `bootstrap-shim` ?

Reply:
No, it's only there so tidy can warn on an outdated version, which doesn't matter for the shim.